corehq/apps/locations/adjacencylist.py

```diff
@@ -178,6 +178,10 @@ def _cte_get_queryset_ancestors(self, node, include_self=False):
     def _cte_get_queryset_descendants(self, *args, **kw):
         return self._cte_get_descendants(*args, **kw)
 
+    def root_nodes(self):
+        parent_col = self.model.parent_id_attr
+        return self.all().filter(**{parent_col + "__isnull": True})
```

Comment:
Thoughts on replacing these with hardcoded `"parent_id"` since it doesn't ever vary?  (At least for those references not in `django_cte`)

Reply:
adjacencylist.py is generic, and the name of the parent id field is customizable. Maybe it's not worth the extra complexity?

Reply:
I think it's more flexibility than is really necessary, but I know MPTT used the same approach.  I don't think it's a big deal either way, and am fine to leave as-is if it's not a nuisance.

Reply:
https://github.com/dimagi/commcare-hq/pull/20707